nixos/modules/services/web-servers/molly-brown.nix

```diff
@@ -0,0 +1,132 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.molly-brown;
+  homeDir = "/var/lib/molly-brown";
+in {
+
+  options.services.molly-brown = {
+
+    enable = mkEnableOption "Enable the Molly-Brown Gemini server.";
+
+    port = mkOption {
+      default = 1965;
+      type = types.int;
+      description = ''
+        TCP port for molly-brown to bind to.
+      '';
+    };
+
+    hostName = mkOption {
+      type = types.str;
+      default = config.networking.hostName;
+      example = "/var/lib/acme/example.com/cert.pem";
+      description = ''
+        The hostname to respond to requests for. Requests for URLs with other
+        hosts will result in a status 53 (PROXY REQUEST REFUSED) response
+      '';
+    };
+
+    certPath = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      example = "/var/lib/acme/example.com/cert.pem";
+      description = ''
+        Path to TLS certificate. If omitted a certificate will be generated
+        automatically.
+      '';
+    };
+
+    keyPath = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      example = "/var/lib/acme/example.com/key.pem";
+      description =
+        "Path to TLS key. If omitted a key will be generated automatically.";
+    };
+
+    docBase = mkOption {
+      type = types.path;
+      default = homeDir + "/docbase";
+      description = "Base directory for Gemini content.";
+    };
+
+    logPath = mkOption {
+      type = types.path;
+      default = homeDir + "/molly.log";
+      description = "Path of molly-brown log file.";
+    };
+
+    extraConfig = mkOption {
```

Comment:
You might consider a `settings` option here instead. Something along these lines:
```
settings = mkOption {
  type = with types; attrsOf (oneOf [ bool int str ]);
  default = {};
};
```
Then you can generate your config file from that and allow a nice structured configuration. This would be in addition to the existing options you have provided.